### PR TITLE
Make getSchedulingLatency use master proxy

### PR DESCRIPTION
**Pull request by gmarek**
instead of SSHing to the machine and curling locally. Depends on #18357

cc @lavalamp

### CODE REVIEW

File: test/e2e/metrics_util.go [hunk 1]
```diff
@@ -257,15 +259,19 @@ func getMetrics(c *client.Client) (string, error) {
...
-	sshResult, err := SSH(cmd, getMasterHost()+":22", testContext.Provider)
-	if err != nil || sshResult.Code != 0 {
-		return result, fmt.Errorf("unexpected error (code: %d) in ssh connection to master: %#v", sshResult.Code, err)
-	}
-	samples, err := extractMetricSamples(sshResult.Stdout)
+	data, err := c.Get().
+		Prefix("proxy").
+		Namespace(api.NamespaceSystem).
+		Resource("pods").
+		Name(fmt.Sprintf("kube-scheduler-%v:%v", testContext.CloudConfig.MasterName, ports.SchedulerPort)).
```
Comment (lavalamp) [inline test/e2e/metrics_util.go]: I expect this to break in GKE (scheduler mirror pod won't be registered with master). Is this run in the GKE tests?

@roberthbailey

File: cluster/gce/config-test.sh [hunk 2]
```diff
@@ -25,7 +25,7 @@ MASTER_DISK_TYPE=pd-ssd
 MASTER_DISK_SIZE=${MASTER_DISK_SIZE:-20GB}
 NODE_DISK_TYPE=${NODE_DISK_TYPE:-pd-standard}
 NODE_DISK_SIZE=${NODE_DISK_SIZE:-100GB}
-REGISTER_MASTER_KUBELET=${REGISTER_MASTER:-false}
+REGISTER_MASTER_KUBELET=${REGISTER_MASTER:-true}
```
Comment (wojtek-t) [inline cluster/gce/config-test.sh]: I'm afraid we can't do it now, right?

Comment (gmarek) [inline cluster/gce/config-test.sh]: Yeah - we need to fix some tests first...

Comment (wojtek-t) [inline cluster/gce/config-test.sh]: So I'm temporary reassigning this PR back to you - feel free to reassing back to me once this is ready for review.

**Comment by gmarek** [comment]
Added a fallback in case master node is not registered (e.g. GKE).

@wojtek-t PTAL

**Comment by gmarek** [comment]
@k8s-bot unit test this

**Comment by wojtek-t** [comment]
LGTM - thanks!

**Comment by gmarek** [comment]
We're OK with exposing scheduler RO endpoint. IIUC @stephenR  https://github.com/kubernetes/kubernetes/pull/18357#discussion_r47363661

**Comment by gmarek** [comment]
@k8s-bot unit test this

**Comment by gmarek** [comment]
@k8s-bot test this

**Pull request merged**